x-pack/plugins/apm/index.js

```diff
@@ -55,6 +55,23 @@ export function apm(kibana) {
     },
 
     init(server) {
+      server.plugins.xpack_main.registerFeature({
+        id: 'apm',
+        name: 'APM',
+        icon: 'apmApp',
+        navLinkId: 'apm',
+        privileges: {
+          all: {
+            app: ['apm'],
```

Comment:
> Do we have any cases where a feature privilege doesn't grant access to the application that defined the feature?

Not that I'm aware of.

> Similarly, do we have any cases where a feature privilege grants access to a different application entirely?

Not yet

> I guess I'm wondering if we need the app construct at all. Can we say that granting a feature privilege automatically grants access to the granting application?

Sure, but we don't necessarily know the "app" which we should be granting access to unless we have the consumer specify it somewhere. The "feature ID" doesn't correlate for discover/dashboard/etc.

Reply:
> Sure, but we don't necessarily know the "app" which we should be granting access to 

Ah yep, that's what's missing. What do you think about moving `app` up a couple of levels (i.e., at the `navLinkId` level), so that they only have to specify it once per feature, rather than once per feature privilege?

Reply:
We definitely can, I was hoping to make it explicit that it was being used for privileges. I still feel weird about us putting the navlink automatically into the actions, but since we're using it for the "space specific apps" I couldn't only put it under privileges and it felt silly making them put it both places. Do you anticipate us needing this for the "space specific apps"?

Reply:
That makes sense. I don't anticipate needing the `app` for space configuration. At this point, I'm planning on using allowing each feature to be toggled, rather than each app, and since that isn't a security feature, I shouldn't have a need to restrict access to the owning application if all features within that app are turned off.

I'm ok leaving it where you have it, I was just wondering out loud if there was a way to simplify

Reply:
I'm with ya, it's rather awkward. At this point, I'm verging on "overly explicit" and redundant, so please call me out when there's other stuff we should consider consolidating.